From 5b29fbd5af9abe5596afa10e4dba2ccb52edddcb Mon Sep 17 00:00:00 2001
From: K Venkata Sneha <vkuppada@qti.qualcomm.com>
Date: Wed, 8 Oct 2025 11:08:37 +0530
Subject: [PATCH] toolchain: Disabling STRIP & RSTRIP for debug builds

Preserve debug symbols by disabling STRIP and RSTRIP for debug builds
is crucial for tools like Valgrind to perform accurate memory analysis
and debugging.

Change-Id: Ic5379046b0d2b7002c03e0390686a60188e5528a
Signed-off-by: K Venkata Sneha <vkuppada@qti.qualcomm.com>
---
 package/libs/toolchain/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/package/libs/toolchain/Makefile b/package/libs/toolchain/Makefile
index 7c117b144d..0fee4bb531 100644
--- a/package/libs/toolchain/Makefile
+++ b/package/libs/toolchain/Makefile
@@ -23,6 +23,11 @@ else
   LIBGCC_VERSION:=<LIBGCC_VERSION>
 endif
 
+ifeq ($(TARGET_VARIANT),debug)
+  STRIP:=:
+  RSTRIP:=:
+endif
+
 define Package/gcc/Default
   SECTION:=libs
   CATEGORY:=Base system
-- 
2.34.1

